datus/tools/func_tool/semantic_tools.py

```
@@ -3,482 +3,516 @@
+
+    # Event loop is running, create a new thread to run the coroutine
+    import threading
+
+    result = None
+    exception = None
+
+    def run_in_thread():
+        nonlocal result, exception
+        try:
+            result = asyncio.run(coro)
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)